PolarisO365.ps1

```diff
@@ -820,170 +821,130 @@ function Get-PolarisO365SharePoint() {
+            usedStorageInBytes
+            effectiveSlaDomain {
+                id
+                name
+            }
+            objectType
+            slaAssignment
+            onDemandSnapshotCount
+        }
+        pageInfo {
+            endCursor
+            hasNextPage
+            hasPreviousPage
+        }
+    }"
+
+    if ($Includes -eq "SitesOnly") {
+        $payload.query = "query O365SharepointObjectQuery(`$after: String, `$o365OrgId:UUID!, `$filter: [Filter!], `$first: Int!, `$sortBy: HierarchySortByField, `$sortOrder: HierarchySortOrder) {
+            $($querySites)
+        }"
+ 
+        $response = Invoke-RestMethod -Method POST -Uri $endpoint -Body $($payload | ConvertTo-JSON -Depth 100) -Headers $headers
+        $o365Sites = $response.data.o365Sites
+    } elseif ($Includes -eq "DocumentLibrariesOnly") {
+        $payload.query = "query o365SharepointDrives(`$after: String, `$o365OrgId:UUID!, `$filter: [Filter!], `$first: Int!, `$sortBy: HierarchySortByField, `$sortOrder: HierarchySortOrder) {
+            $($queryDrives)
         }"
-    }
 
-    if ($Includes -eq "DocumentLibrariesOnly") {
-        $payload.query = "query O365SharepointObjectQuery(`$after: String, `$orgId: UUID!, `$filter: [Filter!]!, `$first: Int!, `$sortBy: HierarchySortByField, `$sortOrder: HierarchySortOrder) {
-            o365SharepointObjects(after: `$after, fid: `$orgId, filter: `$filter, first: `$first, sortBy: `$sortBy, sortOrder: `$sortOrder) {
-                edges {
-                    node {
-                        objectId
-                        parentId
-                        title
-                        ... on O365SharepointDrive {
-                            logicalPath {
-                                fid
-                                name
-                            }
-                            effectiveSlaDomain {
-                                id
-                                name
-                                ... on ClusterSlaDomain {
-                                    cluster {
-                                        id
-                                        name
-                                    }
-                                }
-                            }
-                            objectType
-                            slaAssignment
-                        }
-                    }
-                }
-                pageInfo {
-                    endCursor
-                    hasNextPage
-                    hasPreviousPage
-                }
-            }
+        $response = Invoke-RestMethod -Method POST -Uri $endpoint -Body $($payload | ConvertTo-JSON -Depth 100) -Headers $headers
+        $o365SharepointDrives = $response.data.o365SharepointDrives
+    } else {
+        $payload.query = "query o365SharepointDrives(`$after: String, `$o365OrgId:UUID!, `$filter: [Filter!], `$first: Int!, `$sortBy: HierarchySortByField, `$sortOrder: HierarchySortOrder) {

```

Comment:
Shouldn't we use the query for site not document libraries? The graphQL for elseif case and else case are same

Reply:
The code was updated. Please review the latest version. Thanks.